src/Admin/SiteHealth.php

```diff
@@ -129,28 +155,69 @@ private function get_persistent_object_cache_learn_more_action() {
 	 */
 	public function persistent_object_cache() {
 		$is_using_object_cache = wp_using_ext_object_cache();
-		$data                  = [
+
+		return [
 			'badge'       => [
-				'label' => esc_html__( 'AMP', 'amp' ),
+				'label' => $this->get_badge_label(),
 				'color' => $is_using_object_cache ? 'green' : 'orange',
 			],
 			'description' => esc_html__( 'The AMP plugin performs at its best when persistent object cache is enabled. Object caching is used to more effectively store image dimensions and parsed CSS.', 'amp' ),
 			'actions'     => $this->get_persistent_object_cache_learn_more_action(),
 			'test'        => 'amp_persistent_object_cache',
+			'status'      => $is_using_object_cache ? 'good' : 'recommended',
+			'label'       => $is_using_object_cache
+				? esc_html__( 'Persistent object caching is enabled', 'amp' )
+				: esc_html__( 'Persistent object caching is not enabled', 'amp' ),
 		];
+	}
 
-		$status = $is_using_object_cache ? 'good' : 'recommended';
-		$label  = $is_using_object_cache
-			? __( 'Persistent object caching is enabled', 'amp' )
-			: __( 'Persistent object caching is not enabled', 'amp' );
+	/**
+	 * Gets the test result data for whether the AMP slug (query var) was defined late.
+	 *
+	 * @return array The test data.
+	 */
+	public function slug_definition_timing() {
+		$is_defined_late = $this->amp_slug_customization_watcher->did_customize_late();
 
-		return array_merge(
-			$data,
-			[
-				'status' => $status,
-				'label'  => esc_html( $label ),
-			]
+		$data = [];
+
+		$data['test']  = 'amp_slug_definition_timing';
+		$data['badge'] = [
+			'label' => $this->get_badge_label(),
+			'color' => $is_defined_late ? 'orange' : 'green',
+		];
+
+		$data['status'] = $is_defined_late ? 'recommended' : 'good';
+
+		$data['label'] = $is_defined_late
+			? esc_html__( 'The AMP slug (query var) was defined late', 'amp' )
+			: esc_html__( 'The AMP slug (query var) was defined early', 'amp' );
+
+		$data['description'] = sprintf(
+			/* translators: %s is 'plugins_loaded'  */
+			esc_html__( 'For best results, the AMP slug (query var) should be available early in WordPress\'s execution flow, specifically before the %s action occurs (at priority 4). This slug is used to construct the paired AMP URLs.', 'amp' ),
+			'<code>plugins_loaded</code>'
 		);
+		$data['description'] .= ' ';
+
+		if ( $is_defined_late ) {
+			$data['description'] .= sprintf(
+				/* translators: %1$s is the value of the slug, %2$s is the constant name, %3$s is the filter name */
+				esc_html__( 'Your AMP slug (%1$s) is defined late, most likely in the theme via the %2$s filter or %3$s constant. Make sure this is being defined in a plugin at the top level so it happens before %4$s. While a late-defined AMP slug will still work, it requires extra work to make the slug available earlier in WordPress execution by storing it in an option.', 'amp' ),
```

Comment:
Do we need to inform them the slug is stored in an option? I don't think that's beneficial to the user in any way.

```suggestion
				esc_html__( 'Your AMP slug (%1$s) is defined late, most likely in the theme via the %2$s filter or %3$s constant. Make sure this is being defined in a plugin at the top level so it happens before %4$s. While a late-defined AMP slug will still work, it requires extra work to make the slug available earlier in WordPress execution and is therefore highly not recommended .', 'amp' ),
```

Reply:
My thinking there was to explain how the plugin achieves "hoisting" the custom query var to be made available earlier than it is defined. To me it seems useful to share some of the technical details, as we're already sharing technical details about the filter name and constant name.

Reply:
Hmm yea that makes sense.